packages/core/CalendarManager.ts

```diff
@@ -48,16 +56,29 @@ export const getConnectedCalendars = async (
         }
         const cals = await calendar.listCalendars();
         const calendars = _(cals)
-          .map((cal) => ({
-            ...cal,
-            readOnly: cal.readOnly || false,
-            primary: cal.primary || null,
-            isSelected: selectedCalendars.some((selected) => selected.externalId === cal.externalId),
-            credentialId,
-          }))
+          .map((cal) => {
+            if (cal.primary) {
+              primary = { ...cal, credentialId };
+            }
+            if (cal.externalId === destinationCalendarExternalId) {
+              destinationCalendar = cal;
+            }
+            return {
+              ...cal,
+              readOnly: cal.readOnly || false,
+              primary: cal.primary || null,
+              isSelected: selectedCalendars.some((selected) => selected.externalId === cal.externalId),
+              credentialId,
+            };
+          })
           .sortBy(["primary"])
           .value();
-        const primary = calendars.find((item) => item.primary) ?? calendars.find((cal) => cal !== undefined);
+
+        if (primary && destinationCalendar) {
+          destinationCalendar.primaryEmail = primary.email;
+          destinationCalendar.integrationTitle = integration.title;
+        }
```

Comment:
Once we have the primary and destination calendar, set the integration title and the primary email for that integration. This is better than when we were looping through every calendar a third time to look for these.